pilot/pkg/serviceregistry/serviceentry/servicediscovery.go

```diff
@@ -888,3 +897,26 @@ func makeConfigKey(svc *model.Service) model.ConfigKey {
 		Name:      string(svc.Hostname),
 		Namespace: svc.Attributes.Namespace}
 }
+
+// isHealthy checks that the provided WorkloadEntry is healthy. If health checks are not enabled,
+// it is assumed to always be healthy
+func isHealthy(cfg config.Config) bool {
+	if parseHealthAnnotation(cfg.Annotations[status.WorkloadEntryHealthCheckAnnotation]) {
+		// We default to false if the condition is not set. This ensures newly created WorkloadEntries
+		// are treated as unhealthy until we prove they are healthy by probe success.
+		return status.GetBoolConditionFromSpec(cfg, status.ConditionHealthy, false)
+	}
+	// If health check is not enabled, assume its healthy
+	return true
```

Comment:
It is not correct. We may have some wles that has healthcheck conditions set

Reply:
When would it have health condition set but not the annotation?

We could just make the annotation control the default, but I am not sure there is any case where it would happen?

Reply:
NVM, this feature has not been released in 1.8